File: 백준/Silver/1058. 친구/친구.py
N = int(input())
graph = [list(input()) for _ in range(N)]
visited = [[0 for _ in range(N)] for _ in range(N)]

for k in range(N):
    for i in range(N):
        for j in range(N):
            if i == j:
                continue

            if graph[i][j] == 'Y' or (graph[i][k] == 'Y' and graph[k][j] == 'Y'):
                visited[i][j] = 1

result = 0
for visit in visited:
    result = max(result, sum(visit))
print(result)

File: 백준/Silver/1058. 친구/친구.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")


const n = Number(input[0])
const graph = input.slice(1,1+n).map(row=>row.split(''))
const distance = Array.from({length:n},()=>Array.from({length:n},()=>0))

for(let k=0;k<n;k++){

  for(let i=0;i<n;i++){
    for(let j=0;j<n;j++){
      if(i!==j && (graph[i][j] === 'Y' || (graph[i][k] === 'Y' && graph[k][j] === 'Y'))){
        
        distance[i][j] = 1

      }
    }
  }
}
let answer = 0
distance.forEach((row)=> answer = Math.max(answer,row.reduce((cnt,element)=>cnt + (element),0)))
console.log(answer)